diff --git a/.grit/patterns/terraform_utils.grit b/.grit/patterns/terraform_utils.grit
new file mode 100644
index 000000000..b23959fb9
--- /dev/null
+++ b/.grit/patterns/terraform_utils.grit
@@ -0,0 +1,28 @@
+engine marzano(0.1)
+language hcl
+
+pattern inline_cloudflare_block_to_list($block_name) {
+    block(body=body($body)) where {
+        $first = .,
+        $body <: some bubble($block_name, $first) block(type=`$block_name`, body=$block) as $me where {
+            if ($first <: .) {
+                $first = $me,
+                $block_name += ` =[`,
+            } else {
+                $first += `,
+    {
+    $block
+    }`,
+                $me => .
+            }
+        },
+        // Stick the final closing bracket on the end
+        $first += `]`,
+    }
+}
+
+pattern inline_cloudflare_block_to_map($block_name) {
+    block(type=`$block_name`, body=$block) => `$block_name = {
+  $block
+}`
+}
