diff --git a/internal/services/regional_tiered_cache/testdata/regionaltieredcache.tf b/internal/services/regional_tiered_cache/testdata/regionaltieredcache.tf
new file mode 100644
index 000000000..3720c018d
--- /dev/null
+++ b/internal/services/regional_tiered_cache/testdata/regionaltieredcache.tf
@@ -0,0 +1,5 @@
+
+resource "cloudflare_regional_tiered_cache" "%[1]s" {
+  zone_id = "%[2]s"
+  value   = "%[3]s"
+}
