diff --git a/internal/services/zero_trust_access_application/testdata/accessapplicationzonedomain.tf b/internal/services/zero_trust_access_application/testdata/accessapplicationzonedomain.tf
new file mode 100644
index 000000000..0d28222ab
--- /dev/null
+++ b/internal/services/zero_trust_access_application/testdata/accessapplicationzonedomain.tf
@@ -0,0 +1,13 @@
+
+	resource "cloudflare_zero_trust_access_application" "%[1]s" {
+		zone_id = "%[2]s"
+		name = "%[1]s"
+		domain = "%[1]s.%[3]s"
+	}
+
+	data "cloudflare_zero_trust_access_application" "%[1]s" {
+		zone_id = "%[2]s"
+		domain = "%[1]s.%[3]s"
+		depends_on = [cloudflare_zero_trust_access_application.%[1]s]
+	}
+	
\ No newline at end of file
