diff --git a/internal/services/r2_managed_domain/resource.go b/internal/services/r2_managed_domain/resource.go
index 3fffc0812..b4ee36a8d 100644
--- a/internal/services/r2_managed_domain/resource.go
+++ b/internal/services/r2_managed_domain/resource.go
@@ -12,6 +12,7 @@ import (
 	"github.com/cloudflare/cloudflare-go/v4/option"
 	"github.com/cloudflare/cloudflare-go/v4/r2"
 	"github.com/cloudflare/terraform-provider-cloudflare/internal/apijson"
+	"github.com/cloudflare/terraform-provider-cloudflare/internal/consts"
 	"github.com/cloudflare/terraform-provider-cloudflare/internal/logging"
 	"github.com/hashicorp/terraform-plugin-framework/resource"
 )
@@ -74,6 +75,7 @@ func (r *R2ManagedDomainResource) Create(ctx context.Context, req resource.Creat
 		r2.BucketDomainManagedUpdateParams{
 			AccountID: cloudflare.F(data.AccountID.ValueString()),
 		},
+		option.WithHeader(consts.R2JurisdictionHTTPHeaderName, data.Jurisdiction.ValueString()),
 		option.WithRequestBody("application/json", dataBytes),
 		option.WithResponseBodyInto(&res),
 		option.WithMiddleware(logging.Middleware(ctx)),
@@ -123,6 +125,7 @@ func (r *R2ManagedDomainResource) Update(ctx context.Context, req resource.Updat
 		r2.BucketDomainManagedUpdateParams{
 			AccountID: cloudflare.F(data.AccountID.ValueString()),
 		},
+		option.WithHeader(consts.R2JurisdictionHTTPHeaderName, data.Jurisdiction.ValueString()),
 		option.WithRequestBody("application/json", dataBytes),
 		option.WithResponseBodyInto(&res),
 		option.WithMiddleware(logging.Middleware(ctx)),
