diff --git a/internal/services/zero_trust_access_application/testdata/accessapplicationaccountdomain.tf b/internal/services/zero_trust_access_application/testdata/accessapplicationaccountdomain.tf
new file mode 100644
index 000000000..20135ed6e
--- /dev/null
+++ b/internal/services/zero_trust_access_application/testdata/accessapplicationaccountdomain.tf
@@ -0,0 +1,12 @@
+
+	resource "cloudflare_zero_trust_access_application" "%[1]s" {
+		account_id = "%[2]s"
+		name = "%[1]s"
+		domain = "%[1]s.%[3]s"
+	}
+
+	data "cloudflare_zero_trust_access_application" "%[1]s" {
+		account_id = "%[2]s"
+		domain = "%[1]s.%[3]s"
+		depends_on = [cloudflare_zero_trust_access_application.%[1]s]
+	}
