diff --git a/internal/services/zero_trust_tunnel_cloudflared_token/data_source.go b/internal/services/zero_trust_tunnel_cloudflared_token/data_source.go
index 6fbbc3aae..f6763b402 100644
--- a/internal/services/zero_trust_tunnel_cloudflared_token/data_source.go
+++ b/internal/services/zero_trust_tunnel_cloudflared_token/data_source.go
@@ -64,7 +64,7 @@ func (d *ZeroTrustTunnelCloudflaredTokenDataSource) Read(ctx context.Context, re
 	}
 
 	res := new(http.Response)
-	env := ZeroTrustTunnelCloudflaredTokenResultDataSourceEnvelope{*data}
+	env := ZeroTrustTunnelCloudflaredTokenResultDataSourceEnvelope{}
 	_, err := d.client.ZeroTrust.Tunnels.Cloudflared.Token.Get(
 		ctx,
 		data.TunnelID.ValueString(),
@@ -82,7 +82,7 @@ func (d *ZeroTrustTunnelCloudflaredTokenDataSource) Read(ctx context.Context, re
 		resp.Diagnostics.AddError("failed to deserialize http request", err.Error())
 		return
 	}
-	data = &env.Result
+	data.Token = env.Result
 
 	resp.Diagnostics.Append(resp.State.Set(ctx, &data)...)
 }
