diff --git a/internal/services/zero_trust_access_application/testdata/accessapplicationwithzoneid.tf b/internal/services/zero_trust_access_application/testdata/accessapplicationwithzoneid.tf
new file mode 100644
index 000000000..4cf68109a
--- /dev/null
+++ b/internal/services/zero_trust_access_application/testdata/accessapplicationwithzoneid.tf
@@ -0,0 +1,8 @@
+
+    resource "cloudflare_zero_trust_access_application" "%[1]s" {
+      name    = "%[1]s"
+      zone_id = "%[3]s"
+      domain  = "%[1]s.%[2]s"
+      type    = "self_hosted"
+    }
+  
\ No newline at end of file
