diff --git a/internal/services/zero_trust_tunnel_cloudflared_token/testdata/basic.tf b/internal/services/zero_trust_tunnel_cloudflared_token/testdata/basic.tf
new file mode 100644
index 000000000..fffea5264
--- /dev/null
+++ b/internal/services/zero_trust_tunnel_cloudflared_token/testdata/basic.tf
@@ -0,0 +1,10 @@
+resource "cloudflare_zero_trust_tunnel_cloudflared" "%[2]s" {
+	account_id = "%[1]s"
+	name       = "%[2]s"
+	tunnel_secret     = "AQIDBAUGBwgBAgMEBQYHCAECAwQFBgcIAQIDBAUGBwg="
+}
+
+data "cloudflare_zero_trust_tunnel_cloudflared_token" "%[2]s" {
+	account_id = "%[1]s"
+	tunnel_id  = cloudflare_zero_trust_tunnel_cloudflared.%[2]s.id
+}
