diff --git a/internal/services/custom_hostname/testdata/customhostnamewithcustommetadata.tf b/internal/services/custom_hostname/testdata/customhostnamewithcustommetadata.tf
new file mode 100644
index 000000000..ca7591b17
--- /dev/null
+++ b/internal/services/custom_hostname/testdata/customhostnamewithcustommetadata.tf
@@ -0,0 +1,14 @@
+resource "cloudflare_custom_hostname" "%[2]s" {
+	zone_id = "%[1]s"
+	hostname = "%[2]s.%[3]s"
+	ssl = {
+		method = "txt"
+		type = "dv"
+		wildcard = true
+	}
+	custom_metadata = {
+		"customer_id" = 12345
+		"redirect_to_https" = true
+		"security_tag" = "low"
+	}
+}
