diff --git a/internal/services/origin_ca_certificate/resource.go b/internal/services/origin_ca_certificate/resource.go
index 683e19da4..6f29e10fb 100644
--- a/internal/services/origin_ca_certificate/resource.go
+++ b/internal/services/origin_ca_certificate/resource.go
@@ -106,6 +106,9 @@ func (r *OriginCACertificateResource) Read(ctx context.Context, req resource.Rea
 		return
 	}
 
+	csr := data.Csr
+	requestedValidity := data.RequestedValidity
+
 	res := new(http.Response)
 	env := OriginCACertificateResultEnvelope{*data}
 	_, err := r.client.OriginCACertificates.Get(
@@ -130,6 +133,8 @@ func (r *OriginCACertificateResource) Read(ctx context.Context, req resource.Rea
 		return
 	}
 	data = &env.Result
+	data.Csr = csr
+	data.RequestedValidity = requestedValidity
 
 	resp.Diagnostics.Append(resp.State.Set(ctx, &data)...)
 }
