diff --git a/internal/services/zero_trust_access_service_token/testdata/cloudflareaccessservicetokenbasicconfigwithduration.tf b/internal/services/zero_trust_access_service_token/testdata/cloudflareaccessservicetokenbasicconfigwithduration.tf
new file mode 100644
index 000000000..2abea88cb
--- /dev/null
+++ b/internal/services/zero_trust_access_service_token/testdata/cloudflareaccessservicetokenbasicconfigwithduration.tf
@@ -0,0 +1,6 @@
+
+resource "cloudflare_zero_trust_access_service_token" "%[1]s" {
+  %[3]s_id = "%[4]s"
+  name     = "%[2]s"
+  duration = "%[5]s"
+}
