diff --git a/internal/services/r2_bucket/model.go b/internal/services/r2_bucket/model.go
index 732851fc3..1291c27b1 100644
--- a/internal/services/r2_bucket/model.go
+++ b/internal/services/r2_bucket/model.go
@@ -15,16 +15,22 @@ type R2BucketModel struct {
 	ID           types.String `tfsdk:"id" json:"-,computed"`
 	Name         types.String `tfsdk:"name" json:"name,required"`
 	AccountID    types.String `tfsdk:"account_id" path:"account_id,required"`
-	Location     types.String `tfsdk:"location" json:"locationHint,computed_optional"`
-	StorageClass types.String `tfsdk:"storage_class" json:"storageClass,computed_optional"`
+	Location     types.String `tfsdk:"location" json:"location,computed_optional"`
+	StorageClass types.String `tfsdk:"storage_class" json:"storage_class,computed_optional"`
 	CreationDate types.String `tfsdk:"creation_date" json:"creation_date,computed"`
-	Jurisdiction types.String `tfsdk:"jurisdiction" json:"jurisdiction,computed"`
+	Jurisdiction types.String `tfsdk:"jurisdiction" json:"jurisdiction,computed_optional"`
 }
 
-func (m R2BucketModel) MarshalJSON() (data []byte, err error) {
-	return apijson.MarshalRoot(m)
+type CreateR2BucketModel struct {
+	ID           types.String `tfsdk:"id" json:"-,computed"`
+	Name         types.String `tfsdk:"name" json:"name,required"`
+	AccountID    types.String `tfsdk:"account_id" path:"account_id,required"`
+	Location     types.String `tfsdk:"location" json:"locationHint,computed_optional"`
+	Jurisdiction types.String `tfsdk:"jurisdiction" json:"-,computed_optional"`
+	StorageClass types.String `tfsdk:"storage_class" json:"storageClass,computed_optional"`
+	CreationDate types.String `tfsdk:"creation_date" json:"-,computed"`
 }
 
-func (m R2BucketModel) MarshalJSONForUpdate(state R2BucketModel) (data []byte, err error) {
-	return apijson.MarshalForPatch(m, state)
+func (m CreateR2BucketModel) MarshalJSON() (data []byte, err error) {
+	return apijson.MarshalRoot(m)
 }
