diff --git a/internal/services/hostname_tls_setting/testdata/ciphers.tf b/internal/services/hostname_tls_setting/testdata/ciphers.tf
new file mode 100644
index 000000000..b5737ceda
--- /dev/null
+++ b/internal/services/hostname_tls_setting/testdata/ciphers.tf
@@ -0,0 +1,10 @@
+
+resource "cloudflare_hostname_tls_setting" "%[4]s" {
+	zone_id	= "%[1]s"
+	hostname = "%[2]s"
+	setting_id = "%[3]s"
+	value = [
+    "AES128-SHA",
+    "ECDHE-RSA-AES256-SHA",
+  ]
+}
