diff --git a/internal/services/zero_trust_access_application/testdata/accessapplicationwithinvalidsessionduration.tf b/internal/services/zero_trust_access_application/testdata/accessapplicationwithinvalidsessionduration.tf
new file mode 100644
index 000000000..8fdfb7061
--- /dev/null
+++ b/internal/services/zero_trust_access_application/testdata/accessapplicationwithinvalidsessionduration.tf
@@ -0,0 +1,9 @@
+
+    resource "cloudflare_zero_trust_access_application" "%[1]s" {
+      name             = "%[1]s-updated"
+      zone_id          = "%[3]s"
+      domain           = "%[1]s.%[2]s"
+      type             = "self_hosted"
+      session_duration = "24z"
+  }
+  
\ No newline at end of file
