diff --git a/internal/services/list_item/data_source_model.go b/internal/services/list_item/data_source_model.go
index f5754cb5e..d55a0c410 100644
--- a/internal/services/list_item/data_source_model.go
+++ b/internal/services/list_item/data_source_model.go
@@ -3,12 +3,7 @@
 package list_item
 
 import (
-	"context"
-
-	"github.com/cloudflare/cloudflare-go/v4"
-	"github.com/cloudflare/cloudflare-go/v4/rules"
 	"github.com/cloudflare/terraform-provider-cloudflare/internal/customfield"
-	"github.com/hashicorp/terraform-plugin-framework/diag"
 	"github.com/hashicorp/terraform-plugin-framework/types"
 )
 
@@ -30,13 +25,13 @@ type ListItemDataSourceModel struct {
 	Redirect   customfield.NestedObject[ListItemRedirectDataSourceModel] `tfsdk:"redirect" json:"redirect,computed"`
 }
 
-func (m *ListItemDataSourceModel) toReadParams(_ context.Context) (params rules.ListItemGetParams, diags diag.Diagnostics) {
-	params = rules.ListItemGetParams{
-		AccountID: cloudflare.F(m.AccountID.ValueString()),
-	}
+// func (m *ListItemDataSourceModel) toReadParams(_ context.Context) (params rules.ListItemGetParams, diags diag.Diagnostics) {
+// 	params = rules.ListItemGetParams{
+// 		AccountID: cloudflare.F(m.AccountID.ValueString()),
+// 	}
 
-	return
-}
+// 	return
+// }
 
 type ListItemHostnameDataSourceModel struct {
 	URLHostname types.String `tfsdk:"url_hostname" json:"url_hostname,computed"`
