diff --git a/internal/services/notification_policy/testdata/checkcloudflarenotificationpolicyupdated.tf b/internal/services/notification_policy/testdata/checkcloudflarenotificationpolicyupdated.tf
new file mode 100644
index 000000000..2a20e965d
--- /dev/null
+++ b/internal/services/notification_policy/testdata/checkcloudflarenotificationpolicyupdated.tf
@@ -0,0 +1,12 @@
+
+resource "cloudflare_notification_policy" "%[1]s" {
+  name        = "%[2]s"
+  account_id  = "%[4]s"
+  description = "%[3]s"
+  enabled     =  true
+  alert_type  = "universal_ssl_event_type"
+  mechanisms = {
+    "email": [{"id": "test-updated@example.com"}, {"id": "test2-updated@example.com"}]
+  }
+  filters = {}
+}
\ No newline at end of file
