diff --git a/internal/services/zero_trust_access_group/testdata/accessgroupconfigbasicwithcommonname.tf b/internal/services/zero_trust_access_group/testdata/accessgroupconfigbasicwithcommonname.tf
new file mode 100644
index 000000000..1954be298
--- /dev/null
+++ b/internal/services/zero_trust_access_group/testdata/accessgroupconfigbasicwithcommonname.tf
@@ -0,0 +1,8 @@
+resource "cloudflare_zero_trust_access_group" "%[1]s" {
+  %[2]s_id = "%[3]s"
+  name     = "%[1]s"
+
+  include = [{
+    common_name = { common_name = "common" }
+  }]
+}
