diff --git a/internal/services/ruleset/testdata/rulesetcustomwafbasic.tf b/internal/services/ruleset/testdata/rulesetcustomwafbasic.tf
new file mode 100644
index 000000000..d4c240b8e
--- /dev/null
+++ b/internal/services/ruleset/testdata/rulesetcustomwafbasic.tf
@@ -0,0 +1,14 @@
+
+resource "cloudflare_ruleset" "%[1]s" {
+  zone_id     = "%[3]s"
+  name        = "%[2]s"
+  description = "%[1]s ruleset description"
+  kind        = "zone"
+  phase       = "http_request_firewall_custom"
+
+  rules = [{
+    action      = "challenge"
+    expression  = "(ip.geoip.country eq \"GB\" or ip.geoip.country eq \"FR\") or cf.threat_score > 0"
+    description = "%[1]s ruleset rule description"
+  }]
+}
\ No newline at end of file
