diff --git a/internal/services/page_rule/testdata/pageruleconfigforwardingonly.tf b/internal/services/page_rule/testdata/pageruleconfigforwardingonly.tf
new file mode 100644
index 000000000..1c56d1a11
--- /dev/null
+++ b/internal/services/page_rule/testdata/pageruleconfigforwardingonly.tf
@@ -0,0 +1,12 @@
+
+resource "cloudflare_page_rule" "%[3]s" {
+  zone_id = "%[1]s"
+  target  = "%[2]s"
+  actions = {
+    // on/off options cannot even be set to off without causing error
+    forwarding_url = {
+      url         = "http://%[4]s/forward"
+      status_code = 301
+    }
+  }
+}
