diff --git a/internal/services/zero_trust_access_application/schema.go b/internal/services/zero_trust_access_application/schema.go
index 364b2418a..a9fbd4cb8 100644
--- a/internal/services/zero_trust_access_application/schema.go
+++ b/internal/services/zero_trust_access_application/schema.go
@@ -1253,6 +1253,7 @@ func ResourceSchema(ctx context.Context) schema.Schema {
 				CustomType: customfield.NewNestedObjectType[ZeroTrustAccessApplicationSaaSAppModel](ctx),
 				Attributes: map[string]schema.Attribute{
 					"auth_type": schema.StringAttribute{
+						Computed:    true,
 						Description: "Optional identifier indicating the authentication protocol used for the saas app. Required for OIDC. Default if unset is \"saml\"\nAvailable values: \"saml\", \"oidc\".",
 						Optional:    true,
 						Validators: []validator.String{
@@ -1328,6 +1329,7 @@ func ResourceSchema(ctx context.Context) schema.Schema {
 					},
 					"idp_entity_id": schema.StringAttribute{
 						Description: "The unique identifier for your SaaS application.",
+						Computed:    true,
 						Optional:    true,
 					},
 					"name_id_format": schema.StringAttribute{
@@ -1343,6 +1345,7 @@ func ResourceSchema(ctx context.Context) schema.Schema {
 					},
 					"public_key": schema.StringAttribute{
 						Description: "The Access public certificate that will be used to verify your identity.",
+						Computed:    true,
 						Optional:    true,
 					},
 					"saml_attribute_transform_jsonata": schema.StringAttribute{
@@ -1355,6 +1358,7 @@ func ResourceSchema(ctx context.Context) schema.Schema {
 					},
 					"sso_endpoint": schema.StringAttribute{
 						Description: "The endpoint where your SaaS application will send login requests.",
+						Computed:    true,
 						Optional:    true,
 					},
 					"updated_at": schema.StringAttribute{
@@ -1375,10 +1379,12 @@ func ResourceSchema(ctx context.Context) schema.Schema {
 					},
 					"client_id": schema.StringAttribute{
 						Description: "The application client id",
+						Computed:    true,
 						Optional:    true,
 					},
 					"client_secret": schema.StringAttribute{
 						Description: "The application client secret, only returned on POST request.",
+						Computed:    true,
 						Optional:    true,
 						Sensitive:   true,
 					},
