diff --git a/internal/services/zone/testdata/zonesconfigmatchregex.tf b/internal/services/zone/testdata/zonesconfigmatchregex.tf
new file mode 100644
index 000000000..e934eb847
--- /dev/null
+++ b/internal/services/zone/testdata/zonesconfigmatchregex.tf
@@ -0,0 +1,11 @@
+
+data "cloudflare_zones" "%[2]s" {
+  filter {
+    match  = "baa-*"
+    // This is an ordering fix to ensure that the test suite doesn't assert
+    // state before all the resources are available.
+    paused = "${cloudflare_zone.foo_net.paused}"
+  }
+}
+
+%[1]s
