diff --git a/internal/services/load_balancer_monitor/testdata/loadbalancermonitorconfigemptyexpectedbody.tf b/internal/services/load_balancer_monitor/testdata/loadbalancermonitorconfigemptyexpectedbody.tf
new file mode 100644
index 000000000..de1bed014
--- /dev/null
+++ b/internal/services/load_balancer_monitor/testdata/loadbalancermonitorconfigemptyexpectedbody.tf
@@ -0,0 +1,7 @@
+
+resource "cloudflare_load_balancer_monitor" "%[1]s" {
+  account_id = "%[2]s"
+  expected_body = ""
+  expected_codes = "2xx"
+  description = "we don't want to check for a given body"
+}
\ No newline at end of file
