diff --git a/internal/services/page_rule/testdata/pageruleconfigcachekeyfieldsexcludemultiplevaluesquerystring.tf b/internal/services/page_rule/testdata/pageruleconfigcachekeyfieldsexcludemultiplevaluesquerystring.tf
new file mode 100644
index 000000000..e0d6c0197
--- /dev/null
+++ b/internal/services/page_rule/testdata/pageruleconfigcachekeyfieldsexcludemultiplevaluesquerystring.tf
@@ -0,0 +1,24 @@
+
+resource "cloudflare_page_rule" "%[3]s" {
+  zone_id = "%[1]s"
+  target  = "%[3]s"
+  actions = {
+    cache_key_fields = {
+      cookie = {
+        check_presence = ["cookie_presence"]
+        include        = ["cookie_include"]
+      }
+      header = {
+        check_presence = ["header_presence"]
+        include        = ["header_include"]
+      }
+      host = {
+        resolved = true
+      }
+      query_string = {
+        exclude = ["query1", "query2"]
+      }
+      user = {}
+    }
+  }
+}
