diff --git a/internal/services/zero_trust_access_policy/testdata/accesspolicygeoconfig.tf b/internal/services/zero_trust_access_policy/testdata/accesspolicygeoconfig.tf
new file mode 100644
index 000000000..01bd09d2d
--- /dev/null
+++ b/internal/services/zero_trust_access_policy/testdata/accesspolicygeoconfig.tf
@@ -0,0 +1,17 @@
+resource "cloudflare_zero_trust_access_application" "%[1]s" {
+  name       = "%[1]s"
+  account_id = "%[3]s"
+  domain     = "%[1]s.%[2]s"
+  type       = "self_hosted"
+}
+
+resource "cloudflare_zero_trust_access_policy" "%[1]s" {
+  name           = "%[1]s"
+  account_id     = "%[3]s"
+  decision       = "allow"
+  include = [{
+    geo = {
+      country_code = "US"
+    }
+  }]
+}
