diff --git a/internal/services/ruleset/testdata/rulesetmanagedwafwithoutdescription.tf b/internal/services/ruleset/testdata/rulesetmanagedwafwithoutdescription.tf
new file mode 100644
index 000000000..7d9b141af
--- /dev/null
+++ b/internal/services/ruleset/testdata/rulesetmanagedwafwithoutdescription.tf
@@ -0,0 +1,17 @@
+
+resource "cloudflare_ruleset" "%[1]s" {
+  zone_id = "%[3]s"
+  name    = "%[2]s"
+  kind    = "zone"
+  phase   = "http_request_firewall_managed"
+
+  rules = [{
+    action = "execute"
+    action_parameters = {
+      id = "efb7b8c949ac4650a09736fc376e9aee"
+    }
+    expression  = "true"
+    description = "Execute Cloudflare Managed Ruleset on my zone-level phase ruleset"
+    enabled     = true
+  }]
+}
\ No newline at end of file
