diff --git a/internal/services/zero_trust_tunnel_cloudflared_virtual_network/testdata/tunnelvirtualnetworksimple.tf b/internal/services/zero_trust_tunnel_cloudflared_virtual_network/testdata/tunnelvirtualnetworksimple.tf
new file mode 100644
index 000000000..ceb531a7c
--- /dev/null
+++ b/internal/services/zero_trust_tunnel_cloudflared_virtual_network/testdata/tunnelvirtualnetworksimple.tf
@@ -0,0 +1,6 @@
+resource "cloudflare_zero_trust_tunnel_cloudflared_virtual_network" "%[1]s" {
+	account_id         = "%[3]s"
+	name               = "%[4]s"
+	comment            = "%[2]s"
+	is_default		   = "%[5]t"
+}
