diff --git a/internal/services/dns_zone_transfers_incoming/testdata/incoming.tf b/internal/services/dns_zone_transfers_incoming/testdata/incoming.tf
new file mode 100644
index 000000000..d7dd8b18a
--- /dev/null
+++ b/internal/services/dns_zone_transfers_incoming/testdata/incoming.tf
@@ -0,0 +1,11 @@
+resource "cloudflare_dns_zone_transfers_peer" "%[1]s" {
+  account_id = "%[5]s"
+  name = "%[1]s"
+}
+
+resource "cloudflare_dns_zone_transfers_incoming" "%[1]s" {
+  zone_id = "%[2]s"
+  auto_refresh_seconds = "%[3]d"
+  name = "%[4]s"
+  peers = [cloudflare_dns_zone_transfers_peer.%[1]s.id]
+}
