diff --git a/internal/services/zone_dnssec/custom.go b/internal/services/zone_dnssec/custom.go
new file mode 100644
index 000000000..9a00f843d
--- /dev/null
+++ b/internal/services/zone_dnssec/custom.go
@@ -0,0 +1,27 @@
+package zone_dnssec
+
+import (
+	"context"
+
+	"github.com/cloudflare/cloudflare-go/v4"
+	"github.com/cloudflare/cloudflare-go/v4/dns"
+	"github.com/cloudflare/cloudflare-go/v4/option"
+	"github.com/cloudflare/terraform-provider-cloudflare/internal/logging"
+	"github.com/hashicorp/terraform-plugin-framework/resource"
+)
+
+func disableDNSSECRecord(ctx context.Context, r *ZoneDNSSECResource, zoneID string, resp *resource.DeleteResponse) {
+	var err error
+	_, err = r.client.DNS.DNSSEC.Edit(
+		ctx,
+		dns.DNSSECEditParams{
+			ZoneID: cloudflare.F(zoneID),
+			Status: cloudflare.F[dns.DNSSECEditParamsStatus](dns.DNSSECEditParamsStatusDisabled),
+		},
+		option.WithMiddleware(logging.Middleware(ctx)),
+	)
+	if err != nil {
+		resp.Diagnostics.AddError("failed to make http request", err.Error())
+		return
+	}
+}
