diff --git a/internal/services/api_shield/testdata/apishield.tf b/internal/services/api_shield/testdata/apishield.tf
new file mode 100644
index 000000000..398559cf9
--- /dev/null
+++ b/internal/services/api_shield/testdata/apishield.tf
@@ -0,0 +1,7 @@
+resource "cloudflare_api_shield" "%[1]s" {
+  zone_id = "%[2]s"
+  auth_id_characteristics = [{
+    name = "my-example-header"
+    type = "header"
+  }]
+}
\ No newline at end of file
