diff --git a/internal/services/load_balancer/testdata/loadbalancerconfigadaptiverouting.tf b/internal/services/load_balancer/testdata/loadbalancerconfigadaptiverouting.tf
new file mode 100644
index 000000000..f5e7b4d03
--- /dev/null
+++ b/internal/services/load_balancer/testdata/loadbalancerconfigadaptiverouting.tf
@@ -0,0 +1,10 @@
+resource "cloudflare_load_balancer" "%[3]s" {
+  zone_id = "%[1]s"
+  name = "tf-testacc-lb-adaptive-routing-%[3]s.%[2]s"
+  fallback_pool = "${cloudflare_load_balancer_pool.%[3]s.id}"
+  default_pools = ["${cloudflare_load_balancer_pool.%[3]s.id}"]
+  adaptive_routing = {
+    failover_across_pools = true
+  }
+  session_affinity = "cookie"
+}
