diff --git a/internal/services/custom_hostname/model.go b/internal/services/custom_hostname/model.go
index 2e5f4e6bb..516503c47 100644
--- a/internal/services/custom_hostname/model.go
+++ b/internal/services/custom_hostname/model.go
@@ -38,7 +38,7 @@ func (m CustomHostnameModel) MarshalJSONForUpdate(state CustomHostnameModel) (da
 
 type CustomHostnameSSLModel struct {
 	BundleMethod         types.String                               `tfsdk:"bundle_method" json:"bundle_method,computed_optional"`
-	CertificateAuthority types.String                               `tfsdk:"certificate_authority" json:"certificate_authority,optional"`
+	CertificateAuthority types.String                               `tfsdk:"certificate_authority" json:"certificate_authority,computed_optional"`
 	CloudflareBranding   types.Bool                                 `tfsdk:"cloudflare_branding" json:"cloudflare_branding,optional,no_refresh"`
 	CustomCERTBundle     *[]*CustomHostnameSSLCustomCERTBundleModel `tfsdk:"custom_cert_bundle" json:"custom_cert_bundle,optional,no_refresh"`
 	CustomCertificate    types.String                               `tfsdk:"custom_certificate" json:"custom_certificate,optional"`
