diff --git a/internal/services/load_balancer/model.go b/internal/services/load_balancer/model.go
index 178088db8..c5e6a5d4a 100644
--- a/internal/services/load_balancer/model.go
+++ b/internal/services/load_balancer/model.go
@@ -18,13 +18,13 @@ type LoadBalancerModel struct {
 	FallbackPool              types.String                                                         `tfsdk:"fallback_pool" json:"fallback_pool,required"`
 	Name                      types.String                                                         `tfsdk:"name" json:"name,required"`
 	DefaultPools              *[]types.String                                                      `tfsdk:"default_pools" json:"default_pools,required"`
-	Description               types.String                                                         `tfsdk:"description" json:"description,optional"`
-	SessionAffinityTTL        types.Float64                                                        `tfsdk:"session_affinity_ttl" json:"session_affinity_ttl,optional"`
-	TTL                       types.Float64                                                        `tfsdk:"ttl" json:"ttl,optional"`
-	CountryPools              *map[string]*[]types.String                                          `tfsdk:"country_pools" json:"country_pools,optional"`
-	Networks                  *[]types.String                                                      `tfsdk:"networks" json:"networks,optional"`
-	POPPools                  *map[string]*[]types.String                                          `tfsdk:"pop_pools" json:"pop_pools,optional"`
-	RegionPools               *map[string]*[]types.String                                          `tfsdk:"region_pools" json:"region_pools,optional"`
+	Description               types.String                                                         `tfsdk:"description" json:"description,computed_optional"`
+	SessionAffinityTTL        types.Float64                                                        `tfsdk:"session_affinity_ttl" json:"session_affinity_ttl,computed_optional"`
+	TTL                       types.Float64                                                        `tfsdk:"ttl" json:"ttl,computed_optional"`
+	CountryPools              customfield.Map[customfield.List[types.String]]                      `tfsdk:"country_pools" json:"country_pools,computed_optional"`
+	Networks                  customfield.List[types.String]                                       `tfsdk:"networks" json:"networks,computed_optional"`
+	POPPools                  customfield.Map[customfield.List[types.String]]                      `tfsdk:"pop_pools" json:"pop_pools,computed_optional"`
+	RegionPools               customfield.Map[customfield.List[types.String]]                      `tfsdk:"region_pools" json:"region_pools,computed_optional"`
 	Enabled                   types.Bool                                                           `tfsdk:"enabled" json:"enabled,computed_optional"`
 	Proxied                   types.Bool                                                           `tfsdk:"proxied" json:"proxied,computed_optional"`
 	SessionAffinity           types.String                                                         `tfsdk:"session_affinity" json:"session_affinity,computed_optional"`
@@ -80,18 +80,18 @@ type LoadBalancerRulesFixedResponseModel struct {
 
 type LoadBalancerRulesOverridesModel struct {
 	AdaptiveRouting           customfield.NestedObject[LoadBalancerRulesOverridesAdaptiveRoutingModel]           `tfsdk:"adaptive_routing" json:"adaptive_routing,computed_optional"`
-	CountryPools              *map[string]*[]types.String                                                        `tfsdk:"country_pools" json:"country_pools,optional"`
+	CountryPools              customfield.Map[customfield.List[types.String]]                                    `tfsdk:"country_pools" json:"country_pools,computed_optional"`
 	DefaultPools              *[]types.String                                                                    `tfsdk:"default_pools" json:"default_pools,optional"`
 	FallbackPool              types.String                                                                       `tfsdk:"fallback_pool" json:"fallback_pool,optional"`
 	LocationStrategy          customfield.NestedObject[LoadBalancerRulesOverridesLocationStrategyModel]          `tfsdk:"location_strategy" json:"location_strategy,computed_optional"`
-	POPPools                  *map[string]*[]types.String                                                        `tfsdk:"pop_pools" json:"pop_pools,optional"`
+	POPPools                  customfield.Map[customfield.List[types.String]]                                    `tfsdk:"pop_pools" json:"pop_pools,computed_optional"`
 	RandomSteering            customfield.NestedObject[LoadBalancerRulesOverridesRandomSteeringModel]            `tfsdk:"random_steering" json:"random_steering,computed_optional"`
-	RegionPools               *map[string]*[]types.String                                                        `tfsdk:"region_pools" json:"region_pools,optional"`
+	RegionPools               customfield.Map[customfield.List[types.String]]                                    `tfsdk:"region_pools" json:"region_pools,computed_optional"`
 	SessionAffinity           types.String                                                                       `tfsdk:"session_affinity" json:"session_affinity,computed_optional"`
 	SessionAffinityAttributes customfield.NestedObject[LoadBalancerRulesOverridesSessionAffinityAttributesModel] `tfsdk:"session_affinity_attributes" json:"session_affinity_attributes,computed_optional"`
-	SessionAffinityTTL        types.Float64                                                                      `tfsdk:"session_affinity_ttl" json:"session_affinity_ttl,optional"`
+	SessionAffinityTTL        types.Float64                                                                      `tfsdk:"session_affinity_ttl" json:"session_affinity_ttl,computed_optional"`
 	SteeringPolicy            types.String                                                                       `tfsdk:"steering_policy" json:"steering_policy,computed_optional"`
-	TTL                       types.Float64                                                                      `tfsdk:"ttl" json:"ttl,optional"`
+	TTL                       types.Float64                                                                      `tfsdk:"ttl" json:"ttl,computed_optional"`
 }
 
 type LoadBalancerRulesOverridesAdaptiveRoutingModel struct {
@@ -109,7 +109,7 @@ type LoadBalancerRulesOverridesRandomSteeringModel struct {
 }
 
 type LoadBalancerRulesOverridesSessionAffinityAttributesModel struct {
-	DrainDuration        types.Float64   `tfsdk:"drain_duration" json:"drain_duration,optional"`
+	DrainDuration        types.Float64   `tfsdk:"drain_duration" json:"drain_duration,computed_optional"`
 	Headers              *[]types.String `tfsdk:"headers" json:"headers,optional"`
 	RequireAllHeaders    types.Bool      `tfsdk:"require_all_headers" json:"require_all_headers,computed_optional"`
 	Samesite             types.String    `tfsdk:"samesite" json:"samesite,computed_optional"`
@@ -118,7 +118,7 @@ type LoadBalancerRulesOverridesSessionAffinityAttributesModel struct {
 }
 
 type LoadBalancerSessionAffinityAttributesModel struct {
-	DrainDuration        types.Float64   `tfsdk:"drain_duration" json:"drain_duration,optional"`
+	DrainDuration        types.Float64   `tfsdk:"drain_duration" json:"drain_duration,computed_optional"`
 	Headers              *[]types.String `tfsdk:"headers" json:"headers,optional"`
 	RequireAllHeaders    types.Bool      `tfsdk:"require_all_headers" json:"require_all_headers,computed_optional"`
 	Samesite             types.String    `tfsdk:"samesite" json:"samesite,computed_optional"`
