diff --git a/internal/services/zero_trust_access_application/testdata/accessapplicationconfigwithadefinedtag.tf b/internal/services/zero_trust_access_application/testdata/accessapplicationconfigwithadefinedtag.tf
new file mode 100644
index 000000000..55e4e9d02
--- /dev/null
+++ b/internal/services/zero_trust_access_application/testdata/accessapplicationconfigwithadefinedtag.tf
@@ -0,0 +1,13 @@
+resource "cloudflare_zero_trust_access_tag" "%[1]s" {
+  account_id = "%[4]s"
+  name = "%[1]s"
+}
+
+resource "cloudflare_zero_trust_access_application" "%[1]s" {
+  zone_id                   = "%[2]s"
+  name                      = "%[1]s"
+  domain                    = "%[1]s.%[3]s"
+  type                      = "self_hosted"
+  session_duration          = "24h"
+  tags                      = [cloudflare_zero_trust_access_tag.%[1]s.id]
+}
