diff --git a/internal/services/queue/schema.go b/internal/services/queue/schema.go
index 981bc9b2e..8c95d98a9 100644
--- a/internal/services/queue/schema.go
+++ b/internal/services/queue/schema.go
@@ -36,19 +36,24 @@ func ResourceSchema(ctx context.Context) schema.Schema {
 				Required: true,
 			},
 			"settings": schema.SingleNestedAttribute{
-				Optional: true,
+				Optional:   true,
+				Computed:   true,
+				CustomType: customfield.NewNestedObjectType[QueueSettingsModel](ctx),
 				Attributes: map[string]schema.Attribute{
 					"delivery_delay": schema.Float64Attribute{
 						Description: "Number of seconds to delay delivery of all messages to consumers.",
 						Optional:    true,
+						Computed:    true,
 					},
 					"delivery_paused": schema.BoolAttribute{
 						Description: "Indicates if message delivery to consumers is currently paused.",
 						Optional:    true,
+						Computed:    true,
 					},
 					"message_retention_period": schema.Float64Attribute{
 						Description: "Number of seconds after which an unconsumed message will be delayed.",
 						Optional:    true,
+						Computed:    true,
 					},
 				},
 			},
