diff --git a/internal/services/load_balancer_monitor/testdata/loadbalancermonitorconfigtcpfullyspecified.tf b/internal/services/load_balancer_monitor/testdata/loadbalancermonitorconfigtcpfullyspecified.tf
new file mode 100644
index 000000000..b4d07cafd
--- /dev/null
+++ b/internal/services/load_balancer_monitor/testdata/loadbalancermonitorconfigtcpfullyspecified.tf
@@ -0,0 +1,10 @@
+
+resource "cloudflare_load_balancer_monitor" "test" {
+  account_id = "%[1]s"
+  type = "tcp"
+  timeout = 9
+  interval = 60
+  retries = 5
+  port = 8080
+  description = "this is a very weird tcp load balancer"
+}
