diff --git a/internal/services/zero_trust_device_posture_rule/testdata/devicepostureruleconfiglinuxdistro.tf b/internal/services/zero_trust_device_posture_rule/testdata/devicepostureruleconfiglinuxdistro.tf
new file mode 100644
index 000000000..e700b1f72
--- /dev/null
+++ b/internal/services/zero_trust_device_posture_rule/testdata/devicepostureruleconfiglinuxdistro.tf
@@ -0,0 +1,18 @@
+
+resource "cloudflare_zero_trust_device_posture_rule" "%[1]s" {
+	account_id                = "%[2]s"
+	name                      = "%[1]s"
+	type                      = "os_version"
+	description               = "My description"
+	schedule                  = "24h"
+	expiration                = "24h"
+	match = [{
+		platform = "linux"
+	}]
+	input = {
+  version = "1.0.0"
+        operator = "<"
+		os_distro_name = "ubuntu"
+		os_distro_revision = "1.0.0"
+}
+}
