diff --git a/internal/services/workers_custom_domain/testdata/workerdomainattach.tf b/internal/services/workers_custom_domain/testdata/workerdomainattach.tf
new file mode 100644
index 000000000..9ff7c932b
--- /dev/null
+++ b/internal/services/workers_custom_domain/testdata/workerdomainattach.tf
@@ -0,0 +1,15 @@
+
+# resource "cloudflare_workers_script" "%[1]s-script" {
+#   account_id = "%[3]s"
+#   name = "%[1]s"
+#   content = "%[2]s"
+# }
+
+resource "cloudflare_workers_custom_domain" "%[1]s" {
+	zone_id = "%[5]s"
+	account_id = "%[3]s"
+	hostname = "%[4]s"
+	service = "mute-truth-fdb1"
+	environment = "production"
+	# depends_on = [cloudflare_workers_script.%[1]s-script]
+}
