diff --git a/internal/services/zero_trust_access_application/testdata/accessapplicationconfigwithadefinedidp.tf b/internal/services/zero_trust_access_application/testdata/accessapplicationconfigwithadefinedidp.tf
new file mode 100644
index 000000000..38e834819
--- /dev/null
+++ b/internal/services/zero_trust_access_application/testdata/accessapplicationconfigwithadefinedidp.tf
@@ -0,0 +1,18 @@
+resource "cloudflare_zero_trust_access_identity_provider" "%[1]s" {
+  account_id = "%[4]s"
+  name = "%[1]s"
+  type = "onetimepin"
+  config = {
+    client_id = "test"
+    client_secret = "test"
+  }
+}
+resource "cloudflare_zero_trust_access_application" "%[1]s" {
+  zone_id                   = "%[2]s"
+  name                      = "%[1]s"
+  domain                    = "%[1]s.%[3]s"
+  type                      = "self_hosted"
+  session_duration          = "24h"
+  auto_redirect_to_identity = true
+  allowed_idps              = [cloudflare_zero_trust_access_identity_provider.%[1]s.id]
+}
