diff --git a/internal/services/zero_trust_device_settings/testdata/zt_device_settings_init.tf b/internal/services/zero_trust_device_settings/testdata/zt_device_settings_init.tf
new file mode 100644
index 000000000..c60e6523a
--- /dev/null
+++ b/internal/services/zero_trust_device_settings/testdata/zt_device_settings_init.tf
@@ -0,0 +1,8 @@
+resource "cloudflare_zero_trust_device_settings" "%[1]s" {
+  account_id = "%[2]s"
+  disable_for_time = 20
+  gateway_proxy_enabled = true
+  gateway_udp_proxy_enabled = true
+  root_certificate_installation_enabled = true
+  use_zt_virtual_ip = true
+}
