diff --git a/internal/services/r2_bucket_lifecycle/resource.go b/internal/services/r2_bucket_lifecycle/resource.go
index d10e2f874..686c5b386 100644
--- a/internal/services/r2_bucket_lifecycle/resource.go
+++ b/internal/services/r2_bucket_lifecycle/resource.go
@@ -12,6 +12,7 @@ import (
 	"github.com/cloudflare/cloudflare-go/v4/option"
 	"github.com/cloudflare/cloudflare-go/v4/r2"
 	"github.com/cloudflare/terraform-provider-cloudflare/internal/apijson"
+	"github.com/cloudflare/terraform-provider-cloudflare/internal/consts"
 	"github.com/cloudflare/terraform-provider-cloudflare/internal/logging"
 	"github.com/hashicorp/terraform-plugin-framework/resource"
 )
@@ -74,6 +75,7 @@ func (r *R2BucketLifecycleResource) Create(ctx context.Context, req resource.Cre
 		r2.BucketLifecycleUpdateParams{
 			AccountID: cloudflare.F(data.AccountID.ValueString()),
 		},
+		option.WithHeader(consts.R2JurisdictionHTTPHeaderName, data.Jurisdiction.ValueString()),
 		option.WithRequestBody("application/json", dataBytes),
 		option.WithResponseBodyInto(&res),
 		option.WithMiddleware(logging.Middleware(ctx)),
@@ -123,6 +125,7 @@ func (r *R2BucketLifecycleResource) Update(ctx context.Context, req resource.Upd
 		r2.BucketLifecycleUpdateParams{
 			AccountID: cloudflare.F(data.AccountID.ValueString()),
 		},
+		option.WithHeader(consts.R2JurisdictionHTTPHeaderName, data.Jurisdiction.ValueString()),
 		option.WithRequestBody("application/json", dataBytes),
 		option.WithResponseBodyInto(&res),
 		option.WithMiddleware(logging.Middleware(ctx)),
@@ -159,6 +162,7 @@ func (r *R2BucketLifecycleResource) Read(ctx context.Context, req resource.ReadR
 		r2.BucketLifecycleGetParams{
 			AccountID: cloudflare.F(data.AccountID.ValueString()),
 		},
+		option.WithHeader(consts.R2JurisdictionHTTPHeaderName, data.Jurisdiction.ValueString()),
 		option.WithResponseBodyInto(&res),
 		option.WithMiddleware(logging.Middleware(ctx)),
 	)
