diff --git a/internal/services/load_balancer_pool/data_source_test.go b/internal/services/load_balancer_pool/data_source_test.go
new file mode 100644
index 000000000..4a18ae7fe
--- /dev/null
+++ b/internal/services/load_balancer_pool/data_source_test.go
@@ -0,0 +1,34 @@
+package load_balancer_pool_test
+
+import (
+	"fmt"
+	"os"
+	"testing"
+
+	"github.com/cloudflare/terraform-provider-cloudflare/internal/acctest"
+	"github.com/cloudflare/terraform-provider-cloudflare/internal/utils"
+	"github.com/hashicorp/terraform-plugin-testing/helper/resource"
+)
+
+func TestAccCloudflareLoadBalancerPools(t *testing.T) {
+	accountID := os.Getenv("CLOUDFLARE_ACCOUNT_ID")
+	rnd := utils.GenerateRandomResourceName()
+	name := fmt.Sprintf("data.cloudflare_load_balancer_pools.%s", rnd)
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:                 func() { acctest.TestAccPreCheck(t) },
+		ProtoV6ProviderFactories: acctest.TestAccProtoV6ProviderFactories,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccCloudflareLoadBalancerPoolsConfig(rnd, accountID),
+				Check: resource.ComposeTestCheckFunc(
+					resource.TestCheckResourceAttr(name, "result.#", "2"),
+				),
+			},
+		},
+	})
+}
+
+func testAccCloudflareLoadBalancerPoolsConfig(name, accountID string) string {
+	return acctest.LoadTestCase("loadbalancerpoolsconfig.tf", name, accountID)
+}
