diff --git a/internal/services/workers_kv/model.go b/internal/services/workers_kv/model.go
index 3ade89ae8..13817426e 100644
--- a/internal/services/workers_kv/model.go
+++ b/internal/services/workers_kv/model.go
@@ -19,7 +19,7 @@ type WorkersKVModel struct {
 	KeyName     types.String `tfsdk:"key_name" path:"key_name,required"`
 	AccountID   types.String `tfsdk:"account_id" path:"account_id,required"`
 	NamespaceID types.String `tfsdk:"namespace_id" path:"namespace_id,required"`
-	Metadata    types.String `tfsdk:"metadata" json:"metadata,required,no_refresh"`
+	Metadata    types.String `tfsdk:"metadata" json:"metadata,optional,no_refresh"`
 	Value       types.String `tfsdk:"value" json:"value,required,no_refresh"`
 }
 
