diff --git a/internal/services/zero_trust_access_policy/testdata/accesspolicyreusableconfig.tf b/internal/services/zero_trust_access_policy/testdata/accesspolicyreusableconfig.tf
new file mode 100644
index 000000000..431bedec2
--- /dev/null
+++ b/internal/services/zero_trust_access_policy/testdata/accesspolicyreusableconfig.tf
@@ -0,0 +1,9 @@
+    resource "cloudflare_zero_trust_access_policy" "%[1]s" {
+      name           = "%[1]s"
+      account_id     = "%[2]s"
+      decision       = "allow"
+      include = [{
+        email = ["a@example.com"]
+      }]
+    }
+  
