diff --git a/internal/services/page_rule/model.go b/internal/services/page_rule/model.go
index d58e1f99e..98806aef1 100644
--- a/internal/services/page_rule/model.go
+++ b/internal/services/page_rule/model.go
@@ -3,7 +3,6 @@
 package page_rule
 
 import (
-	"github.com/cloudflare/terraform-provider-cloudflare/internal/apijson"
 	"github.com/hashicorp/terraform-plugin-framework-timetypes/timetypes"
 	"github.com/hashicorp/terraform-plugin-framework/types"
 )
@@ -19,12 +18,15 @@ type PageRuleModel struct {
 	Status     types.String      `tfsdk:"status" json:"status,computed_optional"`
 	CreatedOn  timetypes.RFC3339 `tfsdk:"created_on" json:"created_on,computed" format:"date-time"`
 	ModifiedOn timetypes.RFC3339 `tfsdk:"modified_on" json:"modified_on,computed" format:"date-time"`
+
+	Target  types.String          `tfsdk:"target" json:"target,required"`
+	Actions *PageRuleActionsModel `tfsdk:"actions" json:"actions,required"`
 }
 
 func (m PageRuleModel) MarshalJSON() (data []byte, err error) {
-	return apijson.MarshalRoot(m)
+	return m.marshalCustom()
 }
 
 func (m PageRuleModel) MarshalJSONForUpdate(state PageRuleModel) (data []byte, err error) {
-	return apijson.MarshalForUpdate(m, state)
+	return m.marshalCustomForUpdate(state)
 }
