diff --git a/internal/services/dns_record/testdata/recordconfigapex.tf b/internal/services/dns_record/testdata/recordconfigapex.tf
new file mode 100644
index 000000000..97eb39117
--- /dev/null
+++ b/internal/services/dns_record/testdata/recordconfigapex.tf
@@ -0,0 +1,8 @@
+
+resource "cloudflare_dns_record" "%[2]s" {
+	zone_id = "%[1]s"
+	name = "%[3]s"
+	content = "192.168.0.10"
+	type = "A"
+	ttl = 3600
+}
