diff --git a/internal/services/zero_trust_access_group/testdata/accessgroupconfigexclude.tf b/internal/services/zero_trust_access_group/testdata/accessgroupconfigexclude.tf
new file mode 100644
index 000000000..2cf5a5170
--- /dev/null
+++ b/internal/services/zero_trust_access_group/testdata/accessgroupconfigexclude.tf
@@ -0,0 +1,13 @@
+resource "cloudflare_zero_trust_access_group" "%[1]s" {
+  account_id = "%[2]s"
+  name = "%[1]s"
+
+  include = [{
+    email = { email = "%[3]s" }
+    email_domain = { domain = "example.com" }
+  }]
+
+  exclude = [{
+    email = { email = "%[3]s" }
+  }]
+}
