diff --git a/internal/services/list/testdata/listredirectupdatetargeturl.tf b/internal/services/list/testdata/listredirectupdatetargeturl.tf
new file mode 100644
index 000000000..c63b3a4a8
--- /dev/null
+++ b/internal/services/list/testdata/listredirectupdatetargeturl.tf
@@ -0,0 +1,7 @@
+
+  resource "cloudflare_list" "%[1]s" {
+    account_id = "%[4]s"
+    name = "%[2]s"
+    description = "%[3]s"
+    kind = "redirect"
+  }
