diff --git a/internal/services/zero_trust_access_application/testdata/accessapplicationmisconfiguredcorsallowalloriginswithcredentials.tf b/internal/services/zero_trust_access_application/testdata/accessapplicationmisconfiguredcorsallowalloriginswithcredentials.tf
new file mode 100644
index 000000000..47f52d431
--- /dev/null
+++ b/internal/services/zero_trust_access_application/testdata/accessapplicationmisconfiguredcorsallowalloriginswithcredentials.tf
@@ -0,0 +1,14 @@
+
+    resource "cloudflare_zero_trust_access_application" "%[1]s" {
+      name             = "%[1]s-updated"
+      zone_id          = "%[3]s"
+      domain           = "%[1]s.%[2]s"
+      type             = "self_hosted"
+
+      cors_headers = {
+  allowed_methods = ["GET"]
+        allow_all_origins = true
+        allow_credentials = true
+}
+  }
+  
\ No newline at end of file
