diff --git a/internal/services/magic_wan_static_route/resource.go b/internal/services/magic_wan_static_route/resource.go
index d51f4cd48..06601d3bc 100644
--- a/internal/services/magic_wan_static_route/resource.go
+++ b/internal/services/magic_wan_static_route/resource.go
@@ -65,6 +65,11 @@ func (r *MagicWANStaticRouteResource) Create(ctx context.Context, req resource.C
 	}
 
 	dataBytes, err := data.MarshalJSON()
+
+	// Workaround to wrap the route into an array, which is what the API expects.
+	dataBytes = append([]byte("["), dataBytes...)
+	dataBytes = append(dataBytes, byte(']'))
+
 	if err != nil {
 		resp.Diagnostics.AddError("failed to serialize http request", err.Error())
 		return
