diff --git a/internal/services/certificate_pack/testdata/acccertificatepackadvancedwaitforactiveconfig.tf b/internal/services/certificate_pack/testdata/acccertificatepackadvancedwaitforactiveconfig.tf
new file mode 100644
index 000000000..ae53f22b4
--- /dev/null
+++ b/internal/services/certificate_pack/testdata/acccertificatepackadvancedwaitforactiveconfig.tf
@@ -0,0 +1,13 @@
+
+resource "cloudflare_certificate_pack" "%[3]s" {
+  zone_id = "%[1]s"
+  type = "%[4]s"
+  hosts = [
+    "%[3]s.%[2]s",
+    "%[2]s"
+  ]
+  validation_method = "txt"
+  validity_days = 90
+  certificate_authority = "lets_encrypt"
+  cloudflare_branding = false
+}
