diff --git a/internal/services/waiting_room_settings/resource_test.go b/internal/services/waiting_room_settings/resource_test.go
new file mode 100644
index 000000000..e08be2e94
--- /dev/null
+++ b/internal/services/waiting_room_settings/resource_test.go
@@ -0,0 +1,36 @@
+package waiting_room_settings_test
+
+import (
+	"fmt"
+	"os"
+	"testing"
+
+	"github.com/cloudflare/terraform-provider-cloudflare/internal/acctest"
+	"github.com/cloudflare/terraform-provider-cloudflare/internal/consts"
+	"github.com/cloudflare/terraform-provider-cloudflare/internal/utils"
+	"github.com/hashicorp/terraform-plugin-testing/helper/resource"
+)
+
+func TestAccCloudflareWaitingRoomSettings_Create(t *testing.T) {
+	zoneID := os.Getenv("CLOUDFLARE_ZONE_ID")
+	rnd := utils.GenerateRandomResourceName()
+	name := fmt.Sprintf("cloudflare_waiting_room_settings.%s", rnd)
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:                 func() { acctest.TestAccPreCheck(t) },
+		ProtoV6ProviderFactories: acctest.TestAccProtoV6ProviderFactories,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccCloudflareWaitingRoomSettings(rnd, zoneID),
+				Check: resource.ComposeTestCheckFunc(
+					resource.TestCheckResourceAttr(name, consts.ZoneIDSchemaKey, zoneID),
+					resource.TestCheckResourceAttr(name, "search_engine_crawler_bypass", "true"),
+				),
+			},
+		},
+	})
+}
+
+func testAccCloudflareWaitingRoomSettings(resourceName, zoneID string) string {
+	return acctest.LoadTestCase("waitingroomsettings.tf", resourceName, zoneID)
+}
