diff --git a/internal/services/zero_trust_access_application/testdata/accessapplicationconfigwithhttponlycookieattribute.tf b/internal/services/zero_trust_access_application/testdata/accessapplicationconfigwithhttponlycookieattribute.tf
new file mode 100644
index 000000000..27553dbbd
--- /dev/null
+++ b/internal/services/zero_trust_access_application/testdata/accessapplicationconfigwithhttponlycookieattribute.tf
@@ -0,0 +1,8 @@
+resource "cloudflare_zero_trust_access_application" "%[1]s" {
+  zone_id                    = "%[2]s"
+  name                       = "%[1]s"
+  domain                     = "%[1]s.%[3]s"
+  type                       = "self_hosted"
+  session_duration           = "24h"
+  http_only_cookie_attribute = true
+}
