diff --git a/internal/services/zone_dnssec/resource.go b/internal/services/zone_dnssec/resource.go
index 5e4996d02..c7d1594bf 100644
--- a/internal/services/zone_dnssec/resource.go
+++ b/internal/services/zone_dnssec/resource.go
@@ -194,6 +194,8 @@ func (r *ZoneDNSSECResource) Delete(ctx context.Context, req resource.DeleteRequ
 		return
 	}
 
+	disableDNSSECRecord(ctx, r, data.ZoneID.ValueString(), resp)
+
 	_, err := r.client.DNS.DNSSEC.Delete(
 		ctx,
 		dns.DNSSECDeleteParams{
