diff --git a/internal/services/workers_for_platforms_dispatch_namespace/testdata/workersforplatformsnamespacemanagement.tf b/internal/services/workers_for_platforms_dispatch_namespace/testdata/workersforplatformsnamespacemanagement.tf
new file mode 100644
index 000000000..bf969691e
--- /dev/null
+++ b/internal/services/workers_for_platforms_dispatch_namespace/testdata/workersforplatformsnamespacemanagement.tf
@@ -0,0 +1,5 @@
+
+  resource "cloudflare_workers_for_platforms_dispatch_namespace" "%[1]s" {
+    account_id = "%[2]s"
+    name       = "%[1]s"
+  }
\ No newline at end of file
