diff --git a/internal/services/ruleset/testdata/rulesettransformationruleuripath.tf b/internal/services/ruleset/testdata/rulesettransformationruleuripath.tf
new file mode 100644
index 000000000..de43492ea
--- /dev/null
+++ b/internal/services/ruleset/testdata/rulesettransformationruleuripath.tf
@@ -0,0 +1,23 @@
+
+resource "cloudflare_ruleset" "%[1]s" {
+  zone_id     = "%[3]s"
+  name        = "%[2]s"
+  description = "%[1]s ruleset description"
+  kind        = "zone"
+  phase       = "http_request_transform"
+
+  rules = [{
+    action = "rewrite"
+    action_parameters = {
+      uri = {
+        path = {
+          value = "/static-rewrite"
+        }
+      }
+    }
+
+    expression  = "(http.host eq \"%[4]s\")"
+    description = "URI transformation path example"
+    enabled     = false
+  }]
+}
\ No newline at end of file
