diff --git a/internal/services/hyperdrive_config/testdata/hyperdriveconfigconfigminimum.tf b/internal/services/hyperdrive_config/testdata/hyperdriveconfigconfigminimum.tf
new file mode 100644
index 000000000..c6517ec54
--- /dev/null
+++ b/internal/services/hyperdrive_config/testdata/hyperdriveconfigconfigminimum.tf
@@ -0,0 +1,12 @@
+resource "cloudflare_hyperdrive_config" "%[1]s" {
+	account_id = "%[2]s"
+	name       = "%[3]s"
+	origin     = {
+		password   = "%[4]s"
+		database   = "%[5]s"
+		host       = "%[6]s"
+		port       = "%[7]s"
+		scheme	   = "%[8]s"
+		user       = "%[9]s"
+	}
+}
