diff --git a/internal/services/magic_wan_static_route/model.go b/internal/services/magic_wan_static_route/model.go
index 52351c355..a795bd75f 100644
--- a/internal/services/magic_wan_static_route/model.go
+++ b/internal/services/magic_wan_static_route/model.go
@@ -26,7 +26,7 @@ type MagicWANStaticRouteModel struct {
 	Modified      types.Bool                                                      `tfsdk:"modified" json:"modified,computed,no_refresh"`
 	ModifiedOn    timetypes.RFC3339                                               `tfsdk:"modified_on" json:"modified_on,computed,no_refresh" format:"date-time"`
 	ModifiedRoute customfield.NestedObject[MagicWANStaticRouteModifiedRouteModel] `tfsdk:"modified_route" json:"modified_route,computed,no_refresh"`
-	Route         customfield.NestedObject[MagicWANStaticRouteRouteModel]         `tfsdk:"route" json:"route,computed"`
+	Route         customfield.NestedObject[MagicWANStaticRouteRouteModel]         `tfsdk:"route" json:"route,computed_optional"`
 }
 
 func (m MagicWANStaticRouteModel) MarshalJSON() (data []byte, err error) {
@@ -61,17 +61,17 @@ type MagicWANStaticRouteModifiedRouteScopeModel struct {
 
 type MagicWANStaticRouteRouteModel struct {
 	ID          types.String                                                 `tfsdk:"id" json:"id,computed"`
-	Nexthop     types.String                                                 `tfsdk:"nexthop" json:"nexthop,computed"`
-	Prefix      types.String                                                 `tfsdk:"prefix" json:"prefix,computed"`
-	Priority    types.Int64                                                  `tfsdk:"priority" json:"priority,computed"`
+	Nexthop     types.String                                                 `tfsdk:"nexthop" json:"nexthop,computed_optional"`
+	Prefix      types.String                                                 `tfsdk:"prefix" json:"prefix,computed_optional"`
+	Priority    types.Int64                                                  `tfsdk:"priority" json:"priority,computed_optional"`
 	CreatedOn   timetypes.RFC3339                                            `tfsdk:"created_on" json:"created_on,computed" format:"date-time"`
-	Description types.String                                                 `tfsdk:"description" json:"description,computed"`
+	Description types.String                                                 `tfsdk:"description" json:"description,computed_optional"`
 	ModifiedOn  timetypes.RFC3339                                            `tfsdk:"modified_on" json:"modified_on,computed" format:"date-time"`
-	Scope       customfield.NestedObject[MagicWANStaticRouteRouteScopeModel] `tfsdk:"scope" json:"scope,computed"`
-	Weight      types.Int64                                                  `tfsdk:"weight" json:"weight,computed"`
+	Scope       customfield.NestedObject[MagicWANStaticRouteRouteScopeModel] `tfsdk:"scope" json:"scope,computed_optional"`
+	Weight      types.Int64                                                  `tfsdk:"weight" json:"weight,computed_optional"`
 }
 
 type MagicWANStaticRouteRouteScopeModel struct {
-	ColoNames   customfield.List[types.String] `tfsdk:"colo_names" json:"colo_names,computed"`
-	ColoRegions customfield.List[types.String] `tfsdk:"colo_regions" json:"colo_regions,computed"`
+	ColoNames   customfield.List[types.String] `tfsdk:"colo_names" json:"colo_names,computed_optional"`
+	ColoRegions customfield.List[types.String] `tfsdk:"colo_regions" json:"colo_regions,computed_optional"`
 }
