diff --git a/internal/services/load_balancer/testdata/loadbalancerconfigsessionaffinityipcookie.tf b/internal/services/load_balancer/testdata/loadbalancerconfigsessionaffinityipcookie.tf
new file mode 100644
index 000000000..9154dff53
--- /dev/null
+++ b/internal/services/load_balancer/testdata/loadbalancerconfigsessionaffinityipcookie.tf
@@ -0,0 +1,7 @@
+resource "cloudflare_load_balancer" "%[3]s" {
+  zone_id = "%[1]s"
+  name = "tf-testacc-lb-session-affinity-%[3]s.%[2]s"
+  fallback_pool = "${cloudflare_load_balancer_pool.%[3]s.id}"
+  default_pools = ["${cloudflare_load_balancer_pool.%[3]s.id}"]
+  session_affinity = "ip_cookie"
+}
