diff --git a/internal/services/email_routing_address/testdata/emailroutingaddress.tf b/internal/services/email_routing_address/testdata/emailroutingaddress.tf
new file mode 100644
index 000000000..0e2cf2f97
--- /dev/null
+++ b/internal/services/email_routing_address/testdata/emailroutingaddress.tf
@@ -0,0 +1,5 @@
+
+  resource "cloudflare_email_routing_address" "%[1]s" {
+    account_id = "%[2]s"
+    email      = "%[1]s@example.com"
+  }
\ No newline at end of file
