diff --git a/internal/services/spectrum_application/schema.go b/internal/services/spectrum_application/schema.go
index c03f32553..0638ff4d3 100644
--- a/internal/services/spectrum_application/schema.go
+++ b/internal/services/spectrum_application/schema.go
@@ -111,7 +111,7 @@ func ResourceSchema(ctx context.Context) schema.Schema {
 				Description: "The destination port at the origin. Only specified in conjunction with origin_dns. May use an integer to specify a single origin port, for example `1000`, or a string to specify a range of origin ports, for example `\"1000-2000\"`.\nNotes: If specifying a port range, the number of ports in the range must match the number of ports specified in the \"protocol\" field.",
 				Optional:    true,
 				Validators: []validator.Dynamic{
-					customvalidator.AllowedSubtypes(basetypes.Int64Type{}, basetypes.StringType{}),
+					customvalidator.AllowedSubtypes(basetypes.NumberType{}, basetypes.StringType{}),
 				},
 			},
 			"argo_smart_routing": schema.BoolAttribute{
