diff --git a/internal/services/dns_zone_transfers_outgoing/testdata/outgoing.tf b/internal/services/dns_zone_transfers_outgoing/testdata/outgoing.tf
new file mode 100644
index 000000000..688bc2bec
--- /dev/null
+++ b/internal/services/dns_zone_transfers_outgoing/testdata/outgoing.tf
@@ -0,0 +1,10 @@
+resource "cloudflare_dns_zone_transfers_peer" "%[1]s" {
+  account_id = "%[4]s"
+  name = "%[1]s"
+}
+
+resource "cloudflare_dns_zone_transfers_outgoing" "%[1]s" {
+  zone_id = "%[2]s"
+  name = "%[3]s"
+  peers = [cloudflare_dns_zone_transfers_peer.%[1]s.id]
+}
