diff --git a/internal/services/turnstile_widget/model.go b/internal/services/turnstile_widget/model.go
index fdf4f6cf9..f9f4a9aa9 100644
--- a/internal/services/turnstile_widget/model.go
+++ b/internal/services/turnstile_widget/model.go
@@ -19,10 +19,10 @@ type TurnstileWidgetModel struct {
 	Mode           types.String      `tfsdk:"mode" json:"mode,required"`
 	Name           types.String      `tfsdk:"name" json:"name,required"`
 	Domains        *[]types.String   `tfsdk:"domains" json:"domains,required"`
-	BotFightMode   types.Bool        `tfsdk:"bot_fight_mode" json:"bot_fight_mode,optional"`
-	ClearanceLevel types.String      `tfsdk:"clearance_level" json:"clearance_level,optional"`
-	EphemeralID    types.Bool        `tfsdk:"ephemeral_id" json:"ephemeral_id,optional"`
-	Offlabel       types.Bool        `tfsdk:"offlabel" json:"offlabel,optional"`
+	BotFightMode   types.Bool        `tfsdk:"bot_fight_mode" json:"bot_fight_mode,computed_optional"`
+	ClearanceLevel types.String      `tfsdk:"clearance_level" json:"clearance_level,computed_optional"`
+	EphemeralID    types.Bool        `tfsdk:"ephemeral_id" json:"ephemeral_id,computed_optional"`
+	Offlabel       types.Bool        `tfsdk:"offlabel" json:"offlabel,computed_optional"`
 	Region         types.String      `tfsdk:"region" json:"region,computed_optional"`
 	CreatedOn      timetypes.RFC3339 `tfsdk:"created_on" json:"created_on,computed" format:"date-time"`
 	ModifiedOn     timetypes.RFC3339 `tfsdk:"modified_on" json:"modified_on,computed" format:"date-time"`
