diff --git a/internal/services/custom_hostname_fallback_origin/testdata/customhostnamefallbackorigin.tf b/internal/services/custom_hostname_fallback_origin/testdata/customhostnamefallbackorigin.tf
new file mode 100644
index 000000000..6f178d3ef
--- /dev/null
+++ b/internal/services/custom_hostname_fallback_origin/testdata/customhostnamefallbackorigin.tf
@@ -0,0 +1,15 @@
+
+resource "cloudflare_custom_hostname_fallback_origin" "%[2]s" {
+  zone_id = "%[1]s"
+  origin = "fallback-origin.%[3]s.%[4]s"
+}
+
+resource "cloudflare_dns_record" "%[2]s" {
+  zone_id = "%[1]s"
+  name    = "fallback-origin.%[2]s.%[4]s"
+  content   = "example.com"
+  type    = "CNAME"
+  proxied = true
+  ttl     = 1
+  depends_on = [cloudflare_custom_hostname_fallback_origin.%[2]s]
+}
