diff --git a/internal/services/r2_bucket_lock/model.go b/internal/services/r2_bucket_lock/model.go
index 497ce60cc..d9268f902 100644
--- a/internal/services/r2_bucket_lock/model.go
+++ b/internal/services/r2_bucket_lock/model.go
@@ -13,9 +13,10 @@ type R2BucketLockResultEnvelope struct {
 }
 
 type R2BucketLockModel struct {
-	AccountID  types.String               `tfsdk:"account_id" path:"account_id,required"`
-	BucketName types.String               `tfsdk:"bucket_name" path:"bucket_name,required"`
-	Rules      *[]*R2BucketLockRulesModel `tfsdk:"rules" json:"rules,optional"`
+	AccountID    types.String               `tfsdk:"account_id" path:"account_id,required"`
+	BucketName   types.String               `tfsdk:"bucket_name" path:"bucket_name,required"`
+	Jurisdiction types.String               `tfsdk:"jurisdiction" json:"-,computed_optional"`
+	Rules        *[]*R2BucketLockRulesModel `tfsdk:"rules" json:"rules,optional"`
 }
 
 func (m R2BucketLockModel) MarshalJSON() (data []byte, err error) {
