diff --git a/internal/services/ruleset/testdata/rulesetdisableloggingforskipaction.tf b/internal/services/ruleset/testdata/rulesetdisableloggingforskipaction.tf
new file mode 100644
index 000000000..8b4e8a619
--- /dev/null
+++ b/internal/services/ruleset/testdata/rulesetdisableloggingforskipaction.tf
@@ -0,0 +1,21 @@
+
+resource "cloudflare_ruleset" "%[1]s" {
+  account_id  = "%[3]s"
+  name        = "%[2]s"
+  description = "This ruleset includes a skip rule whose logging is disabled."
+  kind        = "root"
+  phase       = "http_request_firewall_managed"
+
+  rules = [{
+    action = "skip"
+    action_parameters = {
+      ruleset = "current"
+    }
+    expression  = "(cf.zone.plan eq \"ENT\")"
+    enabled     = true
+    description = "example disabled logging"
+    logging = {
+      enabled = false
+    }
+  }]
+}
