diff --git a/internal/services/zero_trust_device_default_profile_local_domain_fallback/testdata/defaultfallbackdomain.tf b/internal/services/zero_trust_device_default_profile_local_domain_fallback/testdata/defaultfallbackdomain.tf
new file mode 100644
index 000000000..2c6ac1587
--- /dev/null
+++ b/internal/services/zero_trust_device_default_profile_local_domain_fallback/testdata/defaultfallbackdomain.tf
@@ -0,0 +1,9 @@
+
+resource "cloudflare_zero_trust_device_default_profile_local_domain_fallback" "%[1]s" {
+  account_id = "%[2]s"
+  domains = [{
+    description = "%[3]s"
+    suffix      = "%[4]s"
+    dns_server  = ["%[5]s"]
+  }]
+}
