diff --git a/internal/services/zero_trust_tunnel_cloudflared/testdata/tunnelmanaged.tf b/internal/services/zero_trust_tunnel_cloudflared/testdata/tunnelmanaged.tf
new file mode 100644
index 000000000..7678c0104
--- /dev/null
+++ b/internal/services/zero_trust_tunnel_cloudflared/testdata/tunnelmanaged.tf
@@ -0,0 +1,7 @@
+
+	resource "cloudflare_zero_trust_tunnel_cloudflared" "%[2]s" {
+		account_id = "%[1]s"
+		name       = "%[2]s"
+		tunnel_secret     = "AQIDBAUGBwgBAgMEBQYHCAECAwQFBgcIAQIDBAUGBwg="
+		config_src = "cloudflare"
+	}
