diff --git a/internal/services/list_item/testdata/hostnameredirectitem.tf b/internal/services/list_item/testdata/hostnameredirectitem.tf
new file mode 100644
index 000000000..5e3c9f574
--- /dev/null
+++ b/internal/services/list_item/testdata/hostnameredirectitem.tf
@@ -0,0 +1,18 @@
+
+  resource "cloudflare_list" "%[2]s" {
+    account_id          = "%[4]s"
+    name                = "%[2]s"
+    description         = "list named %[2]s"
+    kind                = "redirect"
+  }
+
+  resource "cloudflare_list_item" "%[1]s" {
+    account_id = "%[4]s"
+  	list_id    = cloudflare_list.%[2]s.id
+  	comment    = "%[3]s"
+  	redirect = {
+  		source_url = "example.com/"
+  		target_url = "https://example1.com"
+  		status_code = 301
+  	}
+  }
