diff --git a/internal/services/zero_trust_access_application/testdata/data_accessapplicationzonename.tf b/internal/services/zero_trust_access_application/testdata/data_accessapplicationzonename.tf
new file mode 100644
index 000000000..cf694f187
--- /dev/null
+++ b/internal/services/zero_trust_access_application/testdata/data_accessapplicationzonename.tf
@@ -0,0 +1,15 @@
+
+	resource "cloudflare_zero_trust_access_application" "%[1]s" {
+		zone_id = "%[2]s"
+		name = "%[1]s"
+		domain = "%[1]s.%[3]s"
+		type = "self_hosted"
+	}
+
+	data "cloudflare_zero_trust_access_application" "%[1]s" {
+		zone_id = "%[2]s"
+		filter = {
+			name = "%[1]s"
+		}
+		depends_on = [cloudflare_zero_trust_access_application.%[1]s]
+	}
