diff --git a/internal/services/zero_trust_access_group/testdata/accessgroupwithidpauthcontext.tf b/internal/services/zero_trust_access_group/testdata/accessgroupwithidpauthcontext.tf
new file mode 100644
index 000000000..a496879fd
--- /dev/null
+++ b/internal/services/zero_trust_access_group/testdata/accessgroupwithidpauthcontext.tf
@@ -0,0 +1,27 @@
+resource "cloudflare_zero_trust_access_identity_provider" "%[2]s" {
+  account_id = "%[1]s"
+  name = "%[2]s"
+  type = "azureAD"
+  config = {
+    client_id = "test"
+    client_secret = "secret"
+    directory_id = "foo"
+  }
+}
+
+resource "cloudflare_zero_trust_access_group" "%[2]s" {
+  account_id = "%[1]s"
+  name = "%[2]s"
+
+  include = [{
+    any_valid_service_token = {}
+  }]
+
+  require = [{
+    auth_context = {
+      id = "%[3]s"
+      ac_id = "%[4]s"
+      identity_provider_id = cloudflare_zero_trust_access_identity_provider.%[2]s.id
+    }
+  }]
+}
