diff --git a/internal/services/r2_bucket/testdata/r2bucketbasic.tf b/internal/services/r2_bucket/testdata/r2bucketbasic.tf
new file mode 100644
index 000000000..c0f54f1af
--- /dev/null
+++ b/internal/services/r2_bucket/testdata/r2bucketbasic.tf
@@ -0,0 +1,7 @@
+
+  resource "cloudflare_r2_bucket" "%[1]s" {
+    account_id    = "%[2]s"
+    name          = "%[1]s"
+    location      = "ENAM"
+    storage_class = "Standard"
+  }
