diff --git a/internal/services/zero_trust_access_group/testdata/accessgroupconfigemaillist.tf b/internal/services/zero_trust_access_group/testdata/accessgroupconfigemaillist.tf
new file mode 100644
index 000000000..94520c46e
--- /dev/null
+++ b/internal/services/zero_trust_access_group/testdata/accessgroupconfigemaillist.tf
@@ -0,0 +1,16 @@
+resource "cloudflare_zero_trust_list" "%[2]s" {
+  %[3]s_id    = "%[4]s"
+  name        = "%[2]s"
+  type        = "EMAIL"
+  description = "Email list test for %[1]s"
+  items       = [ { value = "test@example.com" } ]
+}
+
+resource "cloudflare_zero_trust_access_group" "%[1]s" {
+  %[3]s_id = "%[4]s"
+  name     = "%[1]s"
+
+  include  = [{
+    email_list = { id = cloudflare_zero_trust_list.%[2]s.id }
+  }]
+}
