diff --git a/internal/services/zero_trust_device_settings/testdata/zt_device_settings_update.tf b/internal/services/zero_trust_device_settings/testdata/zt_device_settings_update.tf
new file mode 100644
index 000000000..26e036ae6
--- /dev/null
+++ b/internal/services/zero_trust_device_settings/testdata/zt_device_settings_update.tf
@@ -0,0 +1,8 @@
+resource "cloudflare_zero_trust_device_settings" "%[1]s" {
+  account_id = "%[2]s"
+  disable_for_time = 0
+  gateway_proxy_enabled = true
+  gateway_udp_proxy_enabled = false
+  root_certificate_installation_enabled = false
+  use_zt_virtual_ip = true
+}
