diff --git a/internal/services/ruleset/testdata/rulesetcachesettingsonlyexludeorigin.tf b/internal/services/ruleset/testdata/rulesetcachesettingsonlyexludeorigin.tf
new file mode 100644
index 000000000..4950245a9
--- /dev/null
+++ b/internal/services/ruleset/testdata/rulesetcachesettingsonlyexludeorigin.tf
@@ -0,0 +1,31 @@
+
+resource "cloudflare_ruleset" "%[1]s" {
+  zone_id     = "%[3]s"
+  name        = "%[2]s"
+  description = "%[1]s ruleset description"
+  kind        = "zone"
+  phase       = "http_request_cache_settings"
+
+  rules = [{
+    action = "set_cache_settings"
+    action_parameters = {
+      edge_ttl = {
+        mode    = "override_origin"
+        default = 60
+      }
+      browser_ttl = {
+        mode = "respect_origin"
+      }
+      cache_key = {
+        custom_key = {
+          header = {
+            exclude_origin = true
+          }
+        }
+      }
+    }
+    expression  = "true"
+    description = "%[1]s set cache settings rule"
+    enabled     = true
+  }]
+}
\ No newline at end of file
