diff --git a/internal/services/web3_hostname/testdata/buildweb3hostnameconfigipfs.tf b/internal/services/web3_hostname/testdata/buildweb3hostnameconfigipfs.tf
new file mode 100644
index 000000000..f4ddffa02
--- /dev/null
+++ b/internal/services/web3_hostname/testdata/buildweb3hostnameconfigipfs.tf
@@ -0,0 +1,8 @@
+
+resource "cloudflare_web3_hostname" "%[1]s" {
+	zone_id = "%[2]s"
+	name = "%[1]s.%[3]s"
+	target = "ipfs"
+	description = "test"
+	dnslink = "/ipns/onboarding.ipfs.cloudflare.com"
+}
