diff --git a/internal/services/firewall_rule/testdata/accfirewallrulewithunicodeandhtmlentityconfig.tf b/internal/services/firewall_rule/testdata/accfirewallrulewithunicodeandhtmlentityconfig.tf
new file mode 100644
index 000000000..eda9af88c
--- /dev/null
+++ b/internal/services/firewall_rule/testdata/accfirewallrulewithunicodeandhtmlentityconfig.tf
@@ -0,0 +1,16 @@
+
+		resource "cloudflare_filter" "%[1]s" {
+		  zone_id = "%[2]s"
+		  paused = "%[3]s"
+		  description = "%[4]s"
+		  expression = "%[5]s"
+		}
+		resource "cloudflare_firewall_rule" "%[1]s" {
+		  zone_id = "%[2]s"
+		  paused = "%[3]s"
+		  description = "%[4]s"
+		  filter_id = "${cloudflare_filter.%[1]s.id}"
+		  action = "%[6]s"
+		  priority = %[7]s
+		}
+		
\ No newline at end of file
