diff --git a/internal/services/zero_trust_access_policy/testdata/accesspolicypurposejustificationconfig.tf b/internal/services/zero_trust_access_policy/testdata/accesspolicypurposejustificationconfig.tf
new file mode 100644
index 000000000..2e2034e1c
--- /dev/null
+++ b/internal/services/zero_trust_access_policy/testdata/accesspolicypurposejustificationconfig.tf
@@ -0,0 +1,20 @@
+resource "cloudflare_zero_trust_access_application" "%[1]s" {
+  name       = "%[1]s"
+  account_id = "%[3]s"
+  domain     = "%[1]s.%[2]s"
+  type       = "self_hosted"
+}
+
+resource "cloudflare_zero_trust_access_policy" "%[1]s" {
+  name           = "%[1]s"
+  account_id     = "%[3]s"
+  decision       = "allow"
+  include = [{
+    email = {
+      email = "a@example.com"
+    }
+  }]
+
+  purpose_justification_required = "true"
+  purpose_justification_prompt = "Why should we let you in?"
+}
