diff --git a/internal/services/zero_trust_access_mtls_certificate/testdata/accessmutualtlscertificateupdated.tf b/internal/services/zero_trust_access_mtls_certificate/testdata/accessmutualtlscertificateupdated.tf
new file mode 100644
index 000000000..207aa23d8
--- /dev/null
+++ b/internal/services/zero_trust_access_mtls_certificate/testdata/accessmutualtlscertificateupdated.tf
@@ -0,0 +1,9 @@
+
+resource "cloudflare_zero_trust_access_mtls_certificate" "%[1]s" {
+	name                 = "%[1]s"
+	%[2]s_id             = "%[3]s"
+	associated_hostnames = []
+	certificate          = <<EOT
+%[4]s
+EOT
+}
