diff --git a/internal/services/zero_trust_access_identity_provider/testdata/accessidentityproviderdatasourcegithub.tf b/internal/services/zero_trust_access_identity_provider/testdata/accessidentityproviderdatasourcegithub.tf
new file mode 100644
index 000000000..b2c914d09
--- /dev/null
+++ b/internal/services/zero_trust_access_identity_provider/testdata/accessidentityproviderdatasourcegithub.tf
@@ -0,0 +1,14 @@
+resource "cloudflare_zero_trust_access_identity_provider" "%[1]s" {
+  account_id = "%[2]s"
+  name = "%[1]s"
+  type = "github"
+  config = {
+    client_id = "test"
+    client_secret = "secret"
+  }
+}
+
+data "cloudflare_zero_trust_access_identity_provider" "%[1]s" {
+  account_id = "%[2]s"
+  depends_on = [cloudflare_zero_trust_access_identity_provider.%[1]s]
+}
