diff --git a/internal/services/load_balancer/testdata/loadbalancerconfigbasic.tf b/internal/services/load_balancer/testdata/loadbalancerconfigbasic.tf
new file mode 100644
index 000000000..ba990cd42
--- /dev/null
+++ b/internal/services/load_balancer/testdata/loadbalancerconfigbasic.tf
@@ -0,0 +1,10 @@
+resource "cloudflare_load_balancer" "%[3]s" {
+  zone_id          = "%[1]s"
+  name             = "tf-testacc-lb-%[3]s.%[2]s"
+  steering_policy  = "off"
+  session_affinity = "none"
+  fallback_pool    = "${cloudflare_load_balancer_pool.%[3]s.id}"
+  default_pools    = [
+    "${cloudflare_load_balancer_pool.%[3]s.id}"
+  ]
+}
