diff --git a/internal/services/zero_trust_tunnel_cloudflared/testdata/default.tf b/internal/services/zero_trust_tunnel_cloudflared/testdata/default.tf
new file mode 100644
index 000000000..68dd31a27
--- /dev/null
+++ b/internal/services/zero_trust_tunnel_cloudflared/testdata/default.tf
@@ -0,0 +1,13 @@
+
+resource "cloudflare_zero_trust_tunnel_cloudflared" "%[2]s" {
+	account_id = "%[1]s"
+	name       = "%[2]s"
+	tunnel_secret     = "AQIDBAUGBwgBAgMEBQYHCAECAwQFBgcIAQIDBAUGBwg="
+}
+
+data "cloudflare_zero_trust_tunnel_cloudflared" "%[2]s" {
+	account_id = "%[1]s"
+	name       = "%[2]s"
+	is_deleted = false
+	depends_on = [cloudflare_zero_trust_tunnel_cloudflared.%[2]s]
+}
