diff --git a/internal/services/byo_ip_prefix/testdata/byoipprefixconfig.tf b/internal/services/byo_ip_prefix/testdata/byoipprefixconfig.tf
new file mode 100644
index 000000000..bdbc0c2d0
--- /dev/null
+++ b/internal/services/byo_ip_prefix/testdata/byoipprefixconfig.tf
@@ -0,0 +1,5 @@
+
+  resource "cloudflare_byo_ip_prefix" "%[3]s" {
+	  prefix_id = "%[1]s"
+	  description = "%[2]s"
+  }
\ No newline at end of file
