diff --git a/internal/services/zero_trust_tunnel_cloudflared_virtual_network/resource.go b/internal/services/zero_trust_tunnel_cloudflared_virtual_network/resource.go
index d7a99ac6e..f890b7480 100644
--- a/internal/services/zero_trust_tunnel_cloudflared_virtual_network/resource.go
+++ b/internal/services/zero_trust_tunnel_cloudflared_virtual_network/resource.go
@@ -112,6 +112,8 @@ func (r *ZeroTrustTunnelCloudflaredVirtualNetworkResource) Update(ctx context.Co
 		return
 	}
 
+	isDefault := state.IsDefault
+
 	dataBytes, err := data.MarshalJSONForUpdate(*state)
 	if err != nil {
 		resp.Diagnostics.AddError("failed to serialize http request", err.Error())
@@ -140,6 +142,7 @@ func (r *ZeroTrustTunnelCloudflaredVirtualNetworkResource) Update(ctx context.Co
 		return
 	}
 	data = &env.Result
+	data.IsDefault = isDefault
 
 	resp.Diagnostics.Append(resp.State.Set(ctx, &data)...)
 }
@@ -153,6 +156,8 @@ func (r *ZeroTrustTunnelCloudflaredVirtualNetworkResource) Read(ctx context.Cont
 		return
 	}
 
+	isDefault := data.IsDefault
+
 	res := new(http.Response)
 	env := ZeroTrustTunnelCloudflaredVirtualNetworkResultEnvelope{*data}
 	_, err := r.client.ZeroTrust.Networks.VirtualNetworks.Get(
@@ -180,6 +185,7 @@ func (r *ZeroTrustTunnelCloudflaredVirtualNetworkResource) Read(ctx context.Cont
 		return
 	}
 	data = &env.Result
+	data.IsDefault = isDefault
 
 	resp.Diagnostics.Append(resp.State.Set(ctx, &data)...)
 }
