diff --git a/internal/services/api_token/testdata/apitokenwithindividualcondition.tf b/internal/services/api_token/testdata/apitokenwithindividualcondition.tf
new file mode 100644
index 000000000..4807db2a9
--- /dev/null
+++ b/internal/services/api_token/testdata/apitokenwithindividualcondition.tf
@@ -0,0 +1,17 @@
+resource "cloudflare_api_token" "%[1]s" {
+	name = "%[1]s"
+
+	policies = [{
+		effect = "allow"
+		permission_groups = [{
+	    id = "%[2]s"
+		}]
+		resources = { "com.cloudflare.api.account.zone.*" = "*" }
+	}]
+
+	condition = {
+    request_ip = {
+			in = ["192.0.2.1/32"]
+		}
+	}
+}
